OpenRA.Mods.Common/ActorIndex.cs

```diff
@@ -111,22 +112,22 @@ public NamesAndTrait(World world, IReadOnlyCollection<string> names)
 
 			static IEnumerable<Actor> ActorsToIndex(World world, HashSet<string> names)
 			{
-				return world.ActorsHavingTrait<T>().Where(a => names.Contains(a.Info.Name));
+				return world.Actors.Where(a => names.Contains(a.Info.Name) && a.Info.HasTraitInfo<TTraitInfo>());
```

Comment:
Why do we use contains over equals?

Reply:
isn't the former check constant time while the latter iterative? Maybe which places?

Reply:
For `names.Contains`? `names` is a collection, so we can't use equals - it's not just one item.